gazebo_ros/src/gazebo_ros_init.cpp

```diff
@@ -162,7 +162,7 @@ void GazeboRosInit::Load(int argc, char ** argv)
   // the simulation is paused), late subscribers can receive the previously published message(s).
   impl_->clock_pub_ = impl_->ros_node_->create_publisher<rosgraph_msgs::msg::Clock>(
     "/clock",
-    rclcpp::QoS(rclcpp::KeepLast(10)).transient_local());
+    rclcpp::ClockQoS());
```

Comment:
I'm having second thoughts about this change; I didn't realize the durability setting for `rclcpp::ClockQoS` is volatile (instead of transient local). I think it might be better to revert this change so we keep the same behavior for anyone depending on the behavior from early versions.